### PR TITLE
validations doc

### PR DESCRIPTION
**Pull request by droslean**
Add document that describe the validation which will take place while the ci-operator is parsing the configuration file.

### CODE REVIEW

File: VALIDATIONS.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# CI Operator Validation Reference
```
Comment (petr-muller) [inline VALIDATIONS.md]: I was told to always call the thing "ci-operator", not "CI Operator" or something similar

File: VALIDATIONS.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+CI-Operator will validate the configuration file with the following rules.
```
Comment (petr-muller) [inline VALIDATIONS.md]: Here too

File: VALIDATIONS.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# ci-operator Validation Reference
+
+ci-operator will validate the configuration file with the following rules.
```
Comment (stevekuznetsov) [inline VALIDATIONS.md]: Nit: we either use "CI Operator" or `ci-operator`.

**Pull request merged**